### PR TITLE
Add a non-ActiveRecord money accessor

### PR DESCRIPTION
**Pull request by tjoyal**
Proposing a new module to support the equivalent of `MoneyColumn::ActiveRecordHooks` for non `ActiveRecord::Base` objects.

```ruby
class NormalObject
  include MoneyAccessor
  
  money_accessor :price

  def initialize(price)
    @price = price
  end
end

object = NormalObject.new(12.34)
object.price # => #<Money value:12.34>
```

The code is somewhat similar to `MoneyColumn::ActiveRecordHooks`, but I believe there was not enough in common to abstract what would be shared. I think it would simply complexify things without any real gain.

### CODE REVIEW

**Comment by tjoyal** [comment]
@davidcornu @sgnr Could I have your opinions and review on this please.

File: spec/money_accessor_spec.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+describe NormalObject do
+  before(:each) do 
+    @object = Object.new
+    @money = Money.new("1.00")
+  end
+
+  it "generate an attribute reader that returns a money object" do
+    object = NormalObject.new(100)
+
+    object.price.should == Money.new(100)
```
Comment (sgnr) [inline spec/money_accessor_spec.rb]: `expect(object.price).to eq(Money.new(100))`

File: lib/money_accessor.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  module ClassMethods
+    def money_accessor(*columns)
+      [columns].flatten.each do |name|
+        define_method(name) do
+          value = instance_variable_get("@#{name}")
+          value.blank? ? nil : Money.new(value)
+        end
+
+        define_method("#{name}=") do |value|
+          if value.blank? || !value.respond_to?(:to_money)
```
Comment (sgnr) [inline lib/money_accessor.rb]: I don't like that it's just discarding it, but I guess the point is to have the exact same behaviour as `ActiveRecordHooks`.

File: spec/money_accessor_spec.rb (hunk 1 above)
Comment (tjoyal) [inline spec/money_accessor_spec.rb]: Done, got rid of the legacy should syntax

File: lib/money_accessor.rb (hunk 2 above)
Comment (tjoyal) [inline lib/money_accessor.rb]: Yep exactly what I had in mind. I prefer everything working the same way.

**Comment by sgnr** [comment]
I think that makes sense. :shipit:

File: lib/money_accessor.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+module MoneyAccessor
+  def self.included(base)
+    base.extend(ClassMethods)
+  end
+
+  module ClassMethods
+    def money_accessor(*columns)
+      [columns].flatten.each do |name|
```
Comment (davidcornu) [inline lib/money_accessor.rb]: `Array(columns).each`?

File: lib/money_column/active_record_hooks.rb [hunk 4]
```diff
@@ -2,14 +2,14 @@ module MoneyColumn
   module ActiveRecordHooks
     def self.included(base)
       base.extend(ClassMethods)
-    end
```
Comment (davidcornu) [inline lib/money_column/active_record_hooks.rb]: facepalm

File: spec/money_accessor_spec.rb [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  end
+end
+
+describe NormalObject do
+  before(:each) do 
+    @object = Object.new
+    @money = Money.new("1.00")
+  end
+
+  it "generate an attribute reader that returns a money object" do
```
Comment (davidcornu) [inline spec/money_accessor_spec.rb]: generate**s**

**Comment by davidcornu** [comment]
Minor stuff. LGTM

**Comment by tjoyal** [comment]
@davidcornu All fixed and :green_apple: !

**Comment by davidcornu** [comment]
:shipit:

**Pull request merged**